tests/cli/cli.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { expect } from "chai";
+import { execFile, fork } from "child_process";
+import { version } from "df/core/version";
+import { suite, test } from "df/testing";
+import * as fs from "fs";
+import * as path from "path";
+
+suite(__filename, () => {
+  const nodePath = "external/nodejs_linux_amd64/bin/node";
+  const npmPath = "external/nodejs_linux_amd64/bin/npm";
+  const corePackageTarPath = "packages/@dataform/core/package.tgz";
+  const cliEntryPointPath = "tests/cli/node_modules/@dataform/cli/bundle.js";
+
+  const projectDir = path.join(process.env.TEST_TMPDIR, "project");
+  const npmCacheDir = path.join(process.env.TEST_TMPDIR, "npm-cache");
+
+  test("init and compile", async () => {
+    // Initialize a project using the CLI, don't install packages.
+    const initChildProcess = fork(cliEntryPointPath, [
+      "init",
+      "redshift",
+      projectDir,
+      "--skip-install"
+    ]);
+
+    await new Promise(resolve => {
+      initChildProcess.on("close", resolve);
+    });
+
+    // Install packages manually to get around bazel sandbox issues.
+    const installChildProcess = execFile(npmPath, [
+      "install",
+      "--prefix",
+      projectDir,
+      "--cache",
+      npmCacheDir,
+      corePackageTarPath
+    ]);
+    installChildProcess.stdout.pipe(process.stdout);
```

Comment:
it's probably worth being consistent and/or wrapping in a (local) utility function how we handle these child processes. (it makes sense to pipe output to this process, but we probably want to do it above too). that function can (a) set the correct pipes up, (b) wait for the child process to complete (with a zero exit code) and (c) return any output it produced